src/components/buttons/async-button.cjsx

```diff
@@ -55,7 +54,7 @@ module.exports = React.createClass
 
     if isFailed or isTimedout
       stateClass = 'is-failed'
-      return <failedState {...failedProps}/>
+      return React.createElement(failedState, failedProps)
```

Comment:
I see your commit message about using `FailedState` instead -- I think we should do that instead of using `React.createElement`.  Using `<FailedState {...failedProps}/>' would be more consistent with our code elsewhere. The jsx would compile to using `React.createElement` anyways.